Python/codegen.c

```diff
@@ -1233,10 +1231,12 @@ codegen_function_body(compiler *c, stmt_ty s, int is_async, Py_ssize_t funcflags
     RETURN_IF_ERROR(
         codegen_enter_scope(c, name, scope_type, (void *)s, firstlineno, NULL, &umd));
 
+    PySTEntryObject *ste = SYMTABLE_ENTRY(c);
     Py_ssize_t first_instr = 0;
     PyObject *docstring = _PyAST_GetDocString(body);
     assert(OPTIMIZATION_LEVEL(c) < 2 || docstring == NULL);
     if (docstring) {
+        assert(ste->ste_has_docstring);
```

Comment:
Is it still true that GetDocstring returns non-null only when ste_has_docstring is true?

Perhaps it should be 
if(ste_has_docstring) assert(GetDocstring)?

Reply:
Currently I think they are equivalent, i.e., `ste_has_docstring == 1` iff. there is a docstring during symbol table generation.

Reply:
And when you run `python -OO`?

Reply:
Currently when running with `-OO`, the first docstring is removed and the subsequent docstring is converted into `JoinedStr`. So both `ste_has_docstring` and `_PyAST_GetDocString` returns 0 for it.

Reply:
Add a test for docstring optimization under `-OO`.

Reply:
> Currently when running with `-OO`, the first docstring is removed and the subsequent docstring is converted into `JoinedStr` in the AST stage. For `_PyAST_GetDocString`, it will not identify the converted `JoinedStr` as docstring, so `ste_has_docstring` will not be set.

Right, but we want to remove that hack.

Reply:
Should we do it in this PR?